src/login/logind-utmp.c

```diff
@@ -43,19 +43,17 @@ _const_ static usec_t when_wall(usec_t n, usec_t elapse) {
 }
 
 bool logind_wall_tty_filter(const char *tty, bool is_local, void *userdata) {
-        Manager *m = userdata;
-        const char *p;
-
-        assert(m);
+        Manager *m = ASSERT_PTR(userdata);
 
-        if (!m->scheduled_shutdown_tty)
-                return true;
-
-        p = path_startswith(tty, "/dev/");
+        const char *p = path_startswith(tty, "/dev/");
         if (!p)
                 return true;
 
-        return !streq(p, m->scheduled_shutdown_tty);
+        /* Do not write to local pseudo-terminals */
+        if (startswith(tty, "pts/") && is_local)
```

Comment:
Arrgh, indeed.

Reply:
Longer term for desktops it would probably make sense to come up with a dbus based notification mechanism the desktops can listen to.  There's already PrepareForShutdown/PrepareForSleep signals but those ones are only sent right before the action takes place.  Unlike the wall message that announce scheduled shutdowns too.